src/test/java/com/learning/selenium/pages/yandex/YandexDiskPage.java

```diff
@@ -0,0 +1,108 @@
+package com.learning.selenium.pages.yandex;
+
+import com.learning.selenium.pages.BasePage;
+import com.learning.selenium.utils.Log;
+import org.openqa.selenium.Keys;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.interactions.Actions;
+import org.openqa.selenium.support.FindBy;
+
+import java.util.List;
+
+public class YandexDiskPage extends BasePage {
+
+  private static final String DISKPAGE_URL = "https://disk.yandex.com/";
+  @FindBy(xpath = "//a[contains(@class, 'button_login')]")
+  private WebElement signInButton;
+  @FindBy(xpath = "//span[@id='/disk']")
+  private WebElement foldersNav;
+  @FindBy(xpath = "//div[@class='listing__items']//div[contains(@class, 'listing-item_type')]")
+  protected List<WebElement> foldersElements;
+  @FindBy(xpath = "//div[contains(@class, 'listing-item_type_dir js-prevent-drag')]")
+  private WebElement trash;
+  private String elementToTrashLocator = "//div[@class='listing__items']//div[contains(@aria-label, '%s')]";
+  @FindBy(xpath = "//div[contains(@class, 'popup__actions-menu')]//div[@role='group']")
+  private List<WebElement> contextMenu;
+  @FindBy(css = "[value=delete]")
+  private WebElement contextMenuDelete;
+
+  public YandexDiskPage(WebDriver driver, Actions actions) {
+    super(driver, actions);
+  }
+
+  @Override
+  public void open() {
+    driver.get(DISKPAGE_URL);
+  }
+
+  public void clickSignInButton() {
+    click(signInButton);
+  }
+
+  public boolean isSignInDisplayed() {
+    return signInButton.isDisplayed();
+  }
+
+  public void dragAndDropIntoTrash(String fileName) {
+    Log.info(String.format("Drag and drop %s into trash", fileName));
+    //WebElement victim = getElement(getLocatorFromOneValue(elementToTrashLocator, fileName));
```

Comment:
please avoid commented code, what's the reason for this?

Reply:
The only reason is that I didn't know will the using dragAndDrop() be enough to complete this task with the actions, so I decided to leave and comment another way to drag and drop.